lesson5.md

```diff
@@ -0,0 +1,543 @@
+
+  console.log("...");
+
+  console.log("General Kenobi!");
+}
+
+sayHello();
+```
+
+---
+
+### Quiz
+
+What does the following code output to console?
+
+```js
+function output42() {
+  console.log("42");
+}
+```
+
+Answer: Nothing, because we never _call_ the function
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+What does the following code output to console?
+
+```js
+function output42() {
+  console.log("42");
+}
+
+output42();
+output42();
+```
+
+Answer: "42" and "42"
+<!-- .element: class="fragment" -->
+
+---
+
+functions can also return some value, we can save that value in a variable to use it later:
+
+```js
+function giveMe5() {
+  return 5;
+}
+
+let number = giveMe5();
+console.log(number); // 5
+```
+
+---
+
+once a function returns, no other code within the function will be executed:
+
+```js
+function giveMe5() {
+  return 5;
+  console.log("I returned 5"); // CODE WILL NEVER EXECUTE!!!
+}
+```
+
+---
+
+A function can have multiple `return` statements. The first `return` statement exits the function:
+
+```js
+function giveMe5Or6() {
+  if (someCondition) {
+    return 5;
+  } else {
+    return 6;
+  }
+  console.log("Done"); // CODE WILL NEVER EXECUTE!!!
+}
+```
+
+---
+
+### Quiz
+
+What is the value of `result`?
+
+```js
+function giveMe5() {
+  return 5;
+}
+
+let result = giveMe5() * giveMe5();
+```
+
+Answer: 25
+<!-- .element: class="fragment" -->
+
+---
+
+functions can also take values, we call them *parameters*:
+
+```js
+function multiplyBy4(x) {
+  return x * 4;
+}
+
+let number = multiplyBy4(7);
+console.log(number); // 28
+
+number = multiplyBy4(number);
+console.log(number);
+// what is the output here?
+```
+
+Answer: 112
+<!-- .element: class="fragment" -->
+
+---
+
+multiple parameters are also possible:
+
+```js
+function multiply(x, y) {
+  return x * y;
+}
+
+let number = multiply(2, 3);
+console.log(number); // 6
+
+number = multiply(10, number);
+console.log(number);
+// what is the output here?
+```
+
+Answer: 60
+<!-- .element: class="fragment" -->
+
+---
+
+### How to create a function
+
+We write:
+* `function`
+* name of the function
+* `(param1, param2, ...)`
+  * parameters are just like variables!
+* `{ ... }` the block where we have our code
+  * we might add a `return` at the end, depending on what we need.
+
+---
+
+```js
+function nameOfFunction (param1, param2, ...) {
+  // any code you want!
+  // even if-else statements!
+  // anything!
+
+  return value; // return is optional, depends on what you need
+
+  // what happens if we don't have a return??
+}
+```
+
+Functions without a return statements return undefined
+<!-- .element: class="fragment" -->
+
+---
+
+### calling a function
+
+to execute a function, we write the function name, and then `()`, we pass any parameters we want into the `()`
+
+```js
+function sayHello(){
+  console.log("Hello!");
+}
+
+// the function sayHello
+// does not require any parameters
+// that is why the parenthesis () are empty.
```

Comment:
```suggestion
to execute a function, we write the function name, and then `()`, we pass any values we want into the `()`

```js
function sayHello(){
  console.log("Hello!");
}

// the function sayHello
// does not require any values, it has no parameters
// that is why the parenthesis () are empty.
```